Piwik.Tracker/PiwikTracker.cs

```diff
@@ -1724,7 +1748,9 @@ protected void SetCookie(string cookieName, string cookieValue, long cookieTtl)
             {
                 return new Dictionary<string, string[]>();
             }
-            return new JavaScriptSerializer().Deserialize<Dictionary<string, string[]>>(HttpUtility.UrlDecode(cookie.Value ?? string.Empty));
+
+            return new JavaScriptSerializer().Deserialize<Dictionary<string, string[]>>(HttpUtility.UrlDecode(cookie.Value ?? string.Empty)) ??

```

Comment:
`Deserialize` returns `null` when deserializing `string.Empty`. The sample code does not check for `null` when using `GetCustomVariablesFromCookie`.